src/main/java/org/opentestsystem/ap/ims/rest/VersionedItemApi.java

```diff
@@ -0,0 +1,54 @@
+package org.opentestsystem.ap.ims.rest;
+
+import lombok.extern.slf4j.Slf4j;
+import org.opentestsystem.ap.common.model.Item;
+import org.opentestsystem.ap.ims.model.ItemResource;
+import org.opentestsystem.ap.ims.service.VersionedItemService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.CacheControl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/v1/versioned-items")
+public class VersionedItemApi {
+    private final VersionedItemService versionedItemService;
+
+    @Autowired
+    public VersionedItemApi(final VersionedItemService versionedItemService) {
+        this.versionedItemService = versionedItemService;
+    }
+
+    @GetMapping("/{itemId}")
+    public Item getItem(@PathVariable final String itemId,
+                        @RequestParam final String version) {
+        return versionedItemService.getItem(itemId, version)
+            .orElseThrow(() ->
+                version == null ?
+                    new IllegalArgumentException("Unable to find an item by id [" + itemId + "]") :
+                    new IllegalArgumentException(
+                        "Unable to find an item by id [" + itemId + "] and version [" + version + "]"));
+    }
+
+    @GetMapping("/{itemId}/resources/{fileName}")
```

Comment:
@bdrainer , this is a "strawman" API to start the discussion. 

I think you know what we need, but just in case: 
- we have an item JSON (returned by the API on line 28)
- we will convert it into QTI XML. While doing it we will need to represent any external resources in some fashion. And the representation of the resources must be TIMS independent.  The XML will be returned to the Item Viewer that will make subsequent calls to get the resources. This assumes that it is enough to just use a `fileName`. My guess is that it is not right but I am not sure what else we will need.

Please advise.

Reply:
Got it.  This makes sense.  I think I can get going on this.  I might use the existing API mentioned a few times, use the existing method to the Item, and add the getAttachedFile method to it.  Small details.

Reply:
It would be great if we could reuse any existing code. I just did not know. Thank you!